pkg/apis/networking/v1alpha1/ingress_validation.go

```diff
@@ -27,7 +27,8 @@ import (
 
 // Validate inspects and validates Ingress object.
 func (i *Ingress) Validate(ctx context.Context) *apis.FieldError {
-	return i.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec")
+	newContext := context.WithValue(ctx, "ingressNamespace", i.Namespace)
```

Comment:
Should we use `apis.WithinParent(apis.WithinSpec(ctx))` instead and fetch the meta below via `ParentMeta`?

Reply:
WithinParent is cool.